### PR TITLE
Testing exercise solutions

### PR DESCRIPTION
**Pull request by mailjatinverma**
solutions to TDD class exercises.

### CODE REVIEW

File: js-exercises/parse-ms/parseMs.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+function parseMs(ms) {
```
Comment (haricharanbole) [inline js-exercises/parse-ms/parseMs.js:1]: Variable name ms can be chosen better. It would be nice to write completely. Same goes with function name parseMs.

File: js-exercises/parse-ms/parseMs.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  // ToDo: reduce magic numbers. use functions/constants.
+  const days = Math.floor(ms / (24 * 60 * 60 * 1000));
+  const hours = Math.floor((ms - (days * 24 * 60 * 60 * 1000)) / 3600000);
+  const minutes = Math.floor((ms - (hours * 60 * 60 * 1000 + days * 24 * 60 * 60 * 1000)) / 60000);
+  const seconds = Math.floor((ms - (minutes * 60 * 1000 + hours * 60 * 60 * 1000 + days * 24 * 60 * 60 * 1000)) / 1000);
+  const milliseconds = ((ms - (seconds * 1000 + minutes * 60 * 1000 + hours * 60 * 60 * 1000 + days * 24 * 60 * 60 * 1000)) / 1);
+  const microseconds = ((ms - (milliseconds + seconds * 1000 + minutes * 60 * 1000 + hours * 60 * 60 * 1000 + days * 24 * 60 * 60 * 1000)) * 1000) / 1;
+  const nanoseconds = (ms - (microseconds * (10 ** 3)+milliseconds + seconds * 1000 + minutes * 60 * 1000 + hours * 60 * 60 * 1000 + days * 24 * 60 * 60 * 1000)) / 10 ** 6;
```
Comment (haricharanbole) [inline js-exercises/parse-ms/parseMs.js:9]: The individual lines have became lengthy. May be, writing separate functions for each will make them short and also enables the for future use!

File: js-exercises/parse-ms/parseMs.test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  test('should return parsed object in various time components', () => {
+    expect(parseMs(1337000001)).toEqual({
+      days: 15,
+      hours: 11,
+      minutes: 23,
+      seconds: 20,
+      milliseconds: 1,
+      microseconds: 0,
+      nanoseconds: 0,
+    });
```
Comment (haricharanbole) [inline js-exercises/parse-ms/parseMs.test.js:16]: Kindly test more corner cases too!

File: js-exercises/update-object/updateObject.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
+function updateObject(indexParam, replacementCharacter, inputArray) {
```
Comment (haricharanbole) [inline js-exercises/update-object/updateObject.js:2]: indexParam can be spelt fully. The function name says it's updateObject but we're taking inputArray. It's contradicting.

File: js-exercises/update-object/updateObject.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  let index = indexParam;
+  if (index >= inputArray.length) {
+    return -1;
+  }
+  if (index < 0) {
+    index = inputArray.length + index;
+  }
+  let outputArray = inputArray;
+  outputArray[index] = replacementCharacter;
+  return outputArray;
```
Comment (haricharanbole) [inline js-exercises/update-object/updateObject.js:12]: This code fails when it receives an object. The problems says it is to update an object. So, please consider that as well.

File: js-exercises/update-object/updateObject.test.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import { updateObject } from './updateObject';
+
+describe('updateObject', () => {
+  test('should return updated object as per given replacement at the provided index of input array', () => {
+    expect(updateObject(3, '_', ['a', 'b', 'c'])).toEqual(-1);
+    expect(updateObject(1, '_', ['a', 'b', 'c'])).toEqual(['a', '_', 'c']);
+    expect(updateObject(-1, '_', ['a', 'b', 'c'])).toEqual(['a', 'b', '_']);
+    expect(updateObject(-1, 'a', ['b', 'c'])).toEqual(['b', 'a']);
+    expect(updateObject(0, 'a', ['b', 'c'])).toEqual(['a', 'c']);
+    expect(updateObject(1, 'a', ['b', 'c'])).toEqual(['b', 'a']);
```
Comment (haricharanbole) [inline js-exercises/update-object/updateObject.test.js:11]: Please test more corner cases too. What if we receive index out of the size of Array or a key not present in object?

**Review by haricharanbole — COMMENTED** [review]
Variables are named decently well but needs some improvement. I'd encourage to write more test cases to test corner cases too.
